mantidproject/mantid
# Fix prompt pulse masking and vanadium peak masking in HRPD

**Description of work.**

A, hopefully, final set of fixes to bring the reduced output of the ISIS powder scripts for HRPD into line with the original scripts in that used mantid 2.x.

It covers:

 - fixing prompt pulse masking time bin values
 - introduce masking of the Vandium bragg peaks 
 - enable solid angle corrections by default

This is marked high priority as HRPD need to verify their reduction very soon to be able to move away from mantid 2.x.

**To test:**

Requires access to the _ISIS_ archive. Also requires the [hrpd_test_data.zip](https://github.com/mantidproject/mantid/files/3481726/hrpd_test_data.zip) set of files. Unzip the test data and fill in the path in the following script:


```python
from isis_powder import hrpd

config_file_path = r"/path/to/downloaded/hrpd_config_example.yaml"
hrpd = hrpd.HRPD(config_file=config_file_path,user_name="test")
hrpd.create_vanadium(first_cycle_run_no="66027", window="30-130", multiple_scattering=False)
hrpd.focus(run_number="66055", window="30-130",vanadium_normalisation = True, do_absorb_corrections=False)
```

It should produce 3 workspaces (plus others) in a group, *66055.raw-ResultTOF_{1,2,3}*. Each should have single spectrum with well-defined peaks like this:

![hrp72694_1-1](https://user-images.githubusercontent.com/733773/62704077-78af1b80-b9e2-11e9-8e51-9c58258cf915.png)

*Note that a custom installer was built for the scientists and they have verified that this now gives much better results.*

Fixes #23731

---

#### Reviewer ####

Please comment on the following ([full description](http://developer.mantidproject.org/ReviewingAPullRequest.html)):

##### Code Review #####

- Is the code of an acceptable quality?
- Does the code conform to the [coding standards](http://developer.mantidproject.org/Standards/)?
- Are the unit tests small and test the class in isolation?
- If there are changes in the release notes then do they describe the changes appropriately?

##### Functional Tests #####

- Do changes function as described? Add comments below that describe the tests performed?
- Do the changes handle unexpected situations, e.g. bad input?
- Has the relevant (user and developer) documentation been added/updated?

Does everything look good? Mark the review as **Approve**. A member of `@mantidproject/gatekeepers` will take care of it.
